cdci_data_analysis/analysis/parameters.py

```diff
@@ -683,14 +661,14 @@ def __init__(self,value=None,E_units='keV',name=None):
         #wtform_dict = {'keV': FloatField}
 
         super(SpectralBoundary, self).__init__(value=value,
-                                   units=E_units,
-                                   check_value=self.check_energy_value,
-                                   name=name,
-                                   allowed_units=_allowed_units)
+                                               units=E_units,
+                                               check_value=self.check_energy_value,
+                                               name=name,allowed_units=_allowed_units)
                                    #wtform_dict=wtform_dict)
 
-
-
+    def get_value_default_format(self, value):
+        # might return int or float
+        return ast.literal_eval(value)
```

Comment:
this should be always float. But actually, this could be deduced from units in `Parameter`. Anything which has physical units is float.

Also there is `Float` class. This can inherit from `Float`. And almost all others, except for `Integer` - that one is int.

this sort of generalization will apply to most if not all cases.

Reply:
Same indeed applies to `Energy` and `DetectionThreshold` classes.